src/arrayMethodReduce.js

```diff
@@ -5,8 +5,28 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+
+    let accumulator;
+
+    if (arguments.length === 2) {
+      accumulator = initialValue;
+
+      findResult(0, this);
+    } else {
+      accumulator = this[0];
+
+      findResult(1, this);
+    }
```

Comment:
Good work, but I prefer using ternary operator instead of if/else statement. You could just make:
```
let accumulator = arguments.length === 2? initialValue : this[0];
const index = arguments.length === 2? 0 : 1;
```
And with that you don't need separate `findResult ` function .